### PR TITLE
feat: add git worktree management

### PR DESCRIPTION
**Pull request by Nek-12**
## Summary
- add first-class git worktree management to Builder with list/create/switch/delete flows
- keep session execution target, runtime cwd, tool cwd, and transcript context aligned to the active worktree
- add smart create-target resolution, worktree reminders, transport/client support, and coverage for worktree lifecycle edge cases

## Testing
- ./scripts/test.sh ./cli/app ./server/sessionlifecycle ./server/transport ./server/worktree ./shared/client ./shared/config ./server/metadata ./server/session -count=1
- ./scripts/build.sh --output ./bin/builder

## Summary by CodeRabbit

* **New Features**
  * Worktree management: /worktree (aliases like /wt) overlay and commands to list, create, switch, and delete Git worktrees; safe checks, optional branch cleanup provenance, setup-script support, and runtime rebinds.
  * Assistant: worktree-mode and worktree-mode-exit prompts with automatic execution-context reminders.
  * Config: new [worktrees] (base_dir, setup_script); default model bumped to gpt-5.5.

* **Documentation**
  * New Worktrees user guide and a documentation-writing standard for public docs.

* **Bug Fixes**
  * TUI height/scroll behavior now supports scrolling on all screens.

### CODE REVIEW

File: server/worktree/service.go [hunk 1]
```diff
@@ -0,0 +1,913 @@
...
+	"builder/server/primaryrun"
+	"builder/server/session"
+	shelltool "builder/server/tools/shell"
+	"builder/shared/clientui"
+	"builder/shared/config"
+	"builder/shared/serverapi"
+	"github.com/google/uuid"
+)
+
+const setupScriptTimeout = 10 * time.Second
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/worktree/service.go]: Not viable - this is for lightweight bring-over work, not full setup, 10 minutes will block the agent's work for a long  time

File: cli/app/commands/commands.go [hunk 2]
```diff
@@ -140,19 +146,31 @@ func (r *Registry) RegisterWithOptions(name string, description string, options
...
+	aliasKey := normalizeCommandName(alias)
+	targetKey := normalizeCommandName(target)
+	if aliasKey == "" || targetKey == "" {
+		return
+	}
+	if aliasKey == targetKey {
+		return
+	}
+	r.aliases[aliasKey] = targetKey
+}
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply cli/app/commands/commands.go:172]: Nit

File: server/runtime/worktree_reminder.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	items := llm.ItemsFromMessages([]llm.Message{message})
+	if strings.TrimSpace(stepID) == "" {
+		return items, nil
+	}
+	if err := e.appendMessage(stepID, message); err != nil {
+		return nil, err
+	}
+	state.HasIssuedInGeneration = true
+	state.IssuedCompactionCount = e.compactionCountSnapshot()
+	if err := e.store.SetWorktreeReminderState(state); err != nil {
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply server/runtime/worktree_reminder.go]: Not applying this as-is. `buildWorktreeReminderRequestItems("")` is used by blank-step request builds in token-count / compaction probe paths, not only by real model turns. Marking the reminder issued there suppresses the next actual worktree reminder and broke `TestSubmitUserMessageInjectsPendingWorktreeEnterReminder` plus the broader `./scripts/test.sh ./cli/app ./shared/config ./server/transport ./server/worktree ./server/metadata ./server/sessionruntime ./server/sessionview ./server/runtime ./server/runtimewire ./server/session -count=1` suite when I tried it. Blank-step path needs to stay read-only.
